src/main/java/com/shop/dao/impl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,215 @@
+            return result > 0;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't delete car with id:" + id, ex);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAll = "SELECT c.id AS car_id, c.model,m.id AS manufacturer_id, m.name "
+                + "AS manufacturer_name, m.country AS manufacturer_country, d.id AS driver_id, "
+                + "d.name AS driver_name, d.license_number FROM cars c\n"
+                + "LEFT JOIN manufacturer m ON m.id = c.manufacturer_id\n"
+                + "LEFT JOIN cars_drivers cd ON cd.car_id = c.id \n"
+                + "LEFT JOIN drivers d ON d.id = cd.driver_id\n"
+                + "WHERE c.deleted = false AND d.id = ? AND d.deleted = false;\n";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getAll,
+                        ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't find cars", ex);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) {
+        try {
+            Long carId = resultSet.getObject("car_id", Long.class);
+            String carModel = resultSet.getString("model");
+            Manufacturer manufacturer = getManufacturer(resultSet);
+            List<Driver> drivers = getDrivers(resultSet, carId);
+            Car car = new Car(carModel, manufacturer);
+            car.setId(carId);
+            car.setDrivers(drivers);
+            return car;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Cant create car from ResultSet", ex);
+        }
+    }
+
+    private Manufacturer getManufacturer(ResultSet resultSet) {
+        try {
+            Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+            String manufacturerName = resultSet.getString("manufacturer_name");
+            String manufacturerCountry = resultSet.getString("manufacturer_country");
+            Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+            manufacturer.setId(manufacturerId);
+            return manufacturer;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't create manufacturer from ResultSet", ex);
+        }
+    }
+
+    private List<Driver> getDrivers(ResultSet resultSet, Long id) {
+        List<Driver> drivers = new ArrayList<>();
+        try {
+            do {
+                Driver driver = getDriver(resultSet);
+                if (driver.getId() != null) {
```

Comment:
```suggestion
                if (driver.getId() != null) {
```
why do we need this check?

Reply:
method getDriver(), can return Driver object with null fields =)

Reply:
@NikolayFatkullin Could you describe such case?